SpaceCore/Patches/EventPatcher.cs

```diff
@@ -46,20 +46,23 @@ public override void Apply(HarmonyInstance harmony, IMonitor monitor)
         /// <summary>The method to call before <see cref="Event.tryEventCommand"/>.</summary>
         private static bool Before_TryEventCommand(Event __instance, GameLocation location, GameTime time, string[] split)
         {
-            var _eventCommandArgs = SpaceCore.Instance.Helper.Reflection.GetField<object[]>(typeof(Event), "_eventCommandArgs").GetValue();
-            var _commandLookup = SpaceCore.Instance.Helper.Reflection.GetField<Dictionary<string, MethodInfo>>(typeof(Event), "_commandLookup").GetValue();
-
-            _eventCommandArgs[0] = location;
-            _eventCommandArgs[1] = time;
-            _eventCommandArgs[2] = split;
             if (split.Length == 0)
                 return false;
-            if (EventPatcher.CustomCommands.ContainsKey(split[0]))
-                EventPatcher.CustomCommands[split[0]].Invoke(null, new[] { __instance, _eventCommandArgs[0], _eventCommandArgs[1], _eventCommandArgs[2] });
-            else if (_commandLookup.ContainsKey(split[0]))
-                _commandLookup[split[0]].Invoke(__instance, _eventCommandArgs);
+            string split0 = split[0];
+
+            object[] _eventCommandArgs = SpaceCore.Reflection.GetField<object[]>(typeof(Event), "_eventCommandArgs").GetValue();
+            var _commandLookup = SpaceCore.Reflection.GetField<Dictionary<string, MethodInfo>>(typeof(Event), "_commandLookup").GetValue();
+
+            if (CustomCommands.ContainsKey(split0))
+                CustomCommands[split0].Invoke(null, new object[] { __instance, location, time, split });
+            else if (_commandLookup.ContainsKey(split0)) {
+                _eventCommandArgs[0] = location;
+                _eventCommandArgs[1] = time;
+                _eventCommandArgs[2] = split;
+                _commandLookup[split0].Invoke(__instance, _eventCommandArgs);
+                }
             else
-                SpaceShared.Log.Warn("ERROR: Invalid command: " + split[0]);
+                SpaceShared.Log.Warn("ERROR: Invalid command: " + split0);
```

Comment:
I try to optimize by reducing indirect refs `split[0]` and not manipulating `_eventCommandArgs` unless necessary.